utils/rpm.mk

```diff
@@ -0,0 +1,103 @@
+# Common definitions for make an RPM, used by both the Agent and the Server.
+#
+# Making a pbench component RPM requires a few steps:
+# 1. Get version number.
+# 2. Update the RPM spec file with that version number etc.
+# 3. Update any GIT submodules.
+# 4. Do a "make install" to a temp directory.
+# 5. Generate a tar ball from the directory.
+# 6. Generate a local SRPM that will be uploaded to COPR for building.
+# 7. Optionally generate a local RPM.
+# 8. Clean up the temp directory
+
+PBENCHTOP := $(shell git rev-parse --show-toplevel)
+TMPDIR = /tmp/opt
+
+prog = pbench-${component}
+VERSION := $(file < ${PBENCHTOP}/${component}/VERSION)
+TBDIR = ${TMPDIR}/${prog}-${VERSION}
+
+RPMDIRS = BUILD BUILDROOT SPECS SOURCES SRPMS RPMS
+
+RPMSRC = ${HOME}/rpmbuild/SOURCES
+RPMSRPM = ${HOME}/rpmbuild/SRPMS
+RPMSPEC = ${HOME}/rpmbuild/SPECS
+
+sha1 := $(shell git rev-parse --short HEAD)
+seqno := $(shell if [ -e ./seqno ] ;then cat ./seqno ;else echo "1" ;fi)
+
+# By default we only build a source RPM
+all: srpm
+
+.PHONY: rpm
+rpm: spec srpm
+	rpmbuild -bb ${RPMSPEC}/${prog}.spec
+
+.PHONY: srpm
+srpm: spec patches tarball
+	rm -f ${RPMSRPM}/$(prog)-*.src.rpm
+	rpmbuild -bs ${RPMSPEC}/${prog}.spec
+
+.PHONY: spec
+spec: rpm-dirs ${prog}.spec.j2
+	if [ -e ./seqno ] ;then expr ${seqno} + 1 > ./seqno ;fi
+	jinja2 ${prog}.spec.j2 -D version=${VERSION} -D gdist=g${sha1} -D seqno=${seqno} > ${RPMSPEC}/${prog}.spec
+	rpmlint ${RPMSPEC}/${prog}.spec
+
+.PHONY: patches
+patches: rpm-dirs
+	if [ -d ./patches ] ;then cp ./patches/* ${RPMSRC} ;fi
+
+.PHONY: tarball
+tarball: rpm-dirs submodules ${subcomps}
+	echo "${sha1}" > ${TBDIR}/${component}/SHA1
+	echo "${seqno}" > ${TBDIR}/${component}/SEQNO
+	tar zcf ${RPMSRC}/${prog}-${VERSION}.tar.gz -C ${TMPDIR} ${prog}-${VERSION}
+	rm -rf ${TMPDIR}
+
+.PHONY: rpm-dirs
+rpm-dirs:
+	mkdir -p $(addprefix ${HOME}/rpmbuild/,${RPMDIRS})
+
+.PHONY: submodules
+submodules:
+	git submodule update --init --recursive
+
+# FIXME:  Do we still need the sed command to correct for Python3?
+#	  (Previous, only the server was doing it, and only for two files.)
+.PHONY: ${subcomps}
+${subcomps}:
+	make -C ${PBENCHTOP}/$@ DESTDIR=${TBDIR}/$@ install
+	# sed -i '1s;.*python$$;#!/usr/bin/env python3;' ${TBDIR}/$@/bin/*
+
+.PHONY: check
+check:
+	if [ "${version}" == "" ] ;then echo "version undefined" > /dev/stderr; exit 1 ;fi
+	if [ "${prog}" == "" ] ;then echo "prog undefined" > /dev/stderr ; exit 2 ;fi
+
+$(RPMSRPM)/$(prog)-$(version)-$(seqno)$(sha1).src.rpm: srpm
+
+ifdef COPR_USER
+_copr_user = ${COPR_USER}
+else
+_copr_user = ${USER}
+endif
+
+COPR_TARGETS = copr copr-test copr-interim copr-index copr-inotify copr-dashboard
+.PHONY: ${COPR_TARGETS}
+${COPR_TARGETS}: $(RPMSRPM)/$(prog)-$(version)-$(seqno)$(sha1).src.rpm
+	copr-cli build $(_copr_user)/$(subst copr,pbench,$@) $(RPMSRPM)/$(prog)-$(VERSION)-$(seqno)g$(sha1).src.rpm
+
+.PHONY: veryclean
+veryclean:: clean rpm-clean
```

Comment:
I'd never even noticed the "veryclean" target, and it does indeed seem rather pointless given the structure.

I like your "cleaner clean" and "veryclean" becomes redundant. But what happened to "dot-clean"? I didn't attempt to trace where those files came from, and whether they're still created...

Reply:
According to the [GNU `make` documentation](https://www.gnu.org/software/make/manual/html_node/Standard-Targets.html#Standard-Targets), "standard" targets are `clean`, `distclean`, `mostlyclean`, and `maintainer-clean`, but [it also mentions](https://www.gnu.org/software/make/manual/html_node/Goals.html#index-clean-_0028standard-target_0029) as "typical" `realclean` and `clobber`.  I'm familiar with `clean`, `distclean`, and `clobber` -- I've never encountered the others.

In any case, after talking with Nick, it sounds like my re-implementation of `rpm-clean` is a little too thorough, so I think I'll replace `distclean` with `clobber` and tone down `rpm-clean` and leave `clean` pointing at it.